app/models/attachment.rb

```diff
@@ -12,16 +12,16 @@ class Attachment < ApplicationRecord
   before_save :nilify_locale_if_blank
   before_save :prevent_saving_of_abstract_base_class
 
-  VALID_COMMAND_PAPER_NUMBER_PREFIXES = ["C.", "Cd.", "Cmd.", "Cmnd.", "Cm."].freeze
+  VALID_COMMAND_PAPER_NUMBER_PREFIXES = ["C.", "Cd.", "Cmd.", "Cmnd.", "Cm.", "CP"].freeze
 
   validates_with AttachmentValidator
   validates :attachable, presence: true
   validates :title, presence: true, length: { maximum: 255 }
   validates :isbn, isbn_format: true, allow_blank: true
   validates :command_paper_number, format: {
-    with: /\A(#{VALID_COMMAND_PAPER_NUMBER_PREFIXES.join('|')}) ?\d+/,
+    with: /\A(#{VALID_COMMAND_PAPER_NUMBER_PREFIXES.map { |prefix| prefix.gsub(/\./, "\\.") }.join('|')}) \d+/,
     allow_blank: true,
```

Comment:
Just curious as to what allow_blank does?

Reply:
It allows `command_paper_number` to be an empty string (""), which would otherwise fail the regex on line 22.